### DISCUSSION TITLE
SNOPT trial version usage

### DISCUSSION THREAD

**Discussion by uttamcadambi07** [mdolab/pyoptsparse#363] [General]
Hi,

I requested for a trial version of SNOPT from the developers and got a pre-compiled one for 3 months.

I have the *.so file with me along with the *.lic license file. I have set up the path to this as per the instructions on the SNOPT website.

However, when I try to use SNOPT with the pyOptSparse framework, I get an error that says "snopt not imported correctly"

I use pyoptsparse with DAFoam and it works well with the IPOPT optimizer. It is version 2.8.1 (pyoptsparse) and snopt version 7. 

Does pyoptsparse reqiore SNOPT be compiled from source? 

```
| pyOptSparse Error: There was an error importing the compiled snopt module    |
+------------------------------------------------------------------------------+

Traceback (most recent call last):
  File "runScript.py", line 184, in <module>
    opt = OPT(args.opt, options=optOptions)
  File "/home/uttamcadambi/dafoam_v2/packages/miniconda3/lib/python3.8/site-packages/pyoptsparse/pyOpt_optimizer.py", line 975, in OPT
    return opt(*args, **kwargs)
  File "/home/uttamcadambi/dafoam_v2/packages/miniconda3/lib/python3.8/site-packages/pyoptsparse/pySNOPT/pySNOPT.py", line 63, in __init__
    raise Error("There was an error importing the compiled snopt module")
pyoptsparse.pyOpt_error.Error: There was an error importing the compiled snopt module
I/O    : uncollated

```

**Comment by ewu63** [comment]
It does not need to be compiled from source, but the `.so` library provided as is cannot be used directly with pyOptSparse. See [here](https://github.com/mdolab/pyoptsparse/issues/335#issuecomment-1690100023) for a way to get around this (note that the link is for macOS but the process should be similar).

**Reply by uttamcadambi07** [reply]
Hi, thanks a lot for the quick response. I put together some information on how to do it based on your suggestion. 
Even after specifying the correct paths, I am getting this error : 
```error: '/dafoam_v2/repos/pyoptsparse-2.8.1/pyoptsparse/pySNOPT/f2py/snoptmodule.c' missing```
I do:
```
python3 -m numpy.f2py -m snopt -c ${SRC}/f2py/snopt.pyf ${SRC}/*.f -L/dafoam_v2/repos/pyoptsparse-2.8.1/pyoptsparse/pySNOPT/source/ -lsnopt7
```
Where SRC is where my libsnopt7.so is located. 

is snoptmodule.c not being generated or something?

I must also mention that the link you provided assumes that the user has the source code (perhaps so does this message also?) I am not sure..

**Reply by uttamcadambi07** [reply]
Fixed this actually. 
https://github.com/mdolab/dafoam/discussions/520#discussion-5872960
You can find it here ^ This is for Linux
